### PR TITLE
Enhamcing the multipath_test.py with new scenarios.

### PR DESCRIPTION
**Pull request by Naresh-ibm**
Enhanced the script with few more scenarios on failing and
re-instating the multipaths.

Signed-off-by: Naresh Bannoth <nbannoth@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by Naresh-ibm** [comment]
[multipath_new_function_added_log.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/1422910/multipath_new_function_added_log.txt)

**Comment by Naresh-ibm** [comment]
this is with reference to #659

File: io/disk/multipath_test.py [hunk 1]
```diff
@@ -103,13 +102,15 @@ def test(self):
...
+            self.log.info("changing Selector policy")
             for policy in ["service-time", "round-robin", "queue-length"]:
                 cmd = "path_selector \"%s 0\"" % policy
                 multipath.form_conf_mpath_file(defaults_extra=cmd)
                 if multipath.get_policy(path_dic["wwid"]) != policy:
                     msg += "%s for %s fails\n" % (policy, path_dic["wwid"])
 
-            # Blacklist wwid
+            # Blacklisting wwid
+            self.log.info("Block listing WWIDs")
```
Comment (narasimhan-v) [inline io/disk/multipath_test.py]: Black

File: io/disk/multipath_test.py [hunk 2]
```diff
@@ -119,12 +120,58 @@ def test(self):
...
+            # Failing and reinstating individual paths eg: sdX
+            self.log.info(" Failing and reinstating the individual paths")
+            print path_dic["paths"]
+            for path in path_dic["paths"]:
+                print path
+                if multipath.fail_path(path) is False:
+                    msg += "test failed while failing %s" % path
+                elif multipath.reinstate_path(path) is False:
+                    msg += "test failed while reinstating %s" % path
+        self.mpath_svc.restart()
```
Comment (narasimhan-v) [inline io/disk/multipath_test.py]: Please use this after the other 2 cases as well, as the need is similar

File: io/disk/multipath_test.py [hunk 3]
```diff
@@ -119,12 +120,58 @@ def test(self):
...
+                if multipath.fail_path(path) is False:
+                    msg += "%s did not failed in short fail" % path
+
+        time.sleep(180)
+        for mpath in self.mpath_list:
+            for path in mpath['paths']:
+                self.log.info("reinstating path %s" % path)
+                self.log.info(" ")
+                if multipath.fail_path(path) is False:
+                    msg += "%s did not recovered  in short fail" % path
```
Comment (narasimhan-v) [inline io/disk/multipath_test.py]: The debug message is same as previous message. Differentiate them.

File: io/disk/multipath_test.py [hunk 4]
```diff
@@ -119,12 +120,58 @@ def test(self):
...
             for disk in path_dic["paths"]:
                 cmd = "devnode %s" % disk
                 multipath.form_conf_mpath_file(blacklist=cmd)
                 if disk in multipath.get_paths(path_dic["wwid"]):
                     msg += "Blacklist of %s fails\n" % disk
+                multipath.form_conf_mpath_file()
+
+            # Failing and reinstating individual paths eg: sdX
+            self.log.info(" Failing and reinstating the individual paths")
+            print path_dic["paths"]
```
Comment (narasimhan-v) [inline io/disk/multipath_test.py]: Is this needed ?

File: io/disk/multipath_test.py [hunk 5]
```diff
@@ -119,12 +120,58 @@ def test(self):
...
+            for path in path_dic["paths"]:
+                print path
```
Comment (narasimhan-v) [inline io/disk/multipath_test.py]: Is this needed ?

File: io/disk/multipath_test.py [hunk 6]
```diff
@@ -119,12 +120,58 @@ def test(self):
...
+                if multipath.fail_path(path) is False:
+                    msg += "test failed while failing %s" % path
+                elif multipath.reinstate_path(path) is False:
+                    msg += "test failed while reinstating %s" % path
+        self.mpath_svc.restart()
+
+        # Failing n-1 paths for short time and reinstating back
+        self.log.info("Failing and reinstating the n-1 paths")
+        for mpath in self.mpath_list:
```
Comment (narasimhan-v) [inline io/disk/multipath_test.py]: Using the existing loop would be nice. Something like this:

```
        for path_dic in self.mpath_list:
            for path in path_dic["paths"]:
                if multipath.fail_path(path) is False:
                    msg += "test failed while failing %s" % path
                elif multipath.reinstate_path(path) is False:
                    msg += "test failed while reinstating %s" % path
                for mpath in self.mpath_list:
            self.mpath_svc.restart()
            for path in path_dic['paths'][:-1]:
                self.log.info("failing path %s" % path)
                ...
                time.sleep(180)
            self.mpath_svc.restart()
            for path in path_dic['paths'][:-1]:
                self.log.info("reinstating path %s" % path)
```

**Comment by narasimhan-v** [comment]
Not sure why this is left open.

**Pull request closed without merging**